### PR TITLE
fix some bugs in camera.py

### LINKED ISSUE #2936: pygame.camera members not defined before init?
I tried to call pygame.camera.list_cameras() and it said 

`AttributeError: module 'pygame.camera' has no attribute 'list_cameras'`

When it should have complained that I hadn't initialized the camera module yet, which was the real problem.

This bug is 100% something I caused, but I'm putting up an issue in case someone wants to take a crack at it before I do. Code in src_py/camera.py

### PR DESCRIPTION
**Pull request by ankith26**
Initially set out to fix #2936 then noticed a few more issues, got sidetracked as usual and now I make this PR with a few more fixes 

Now all functions are defined at all times regardless init state, and error correctly. After a quit, they are reset back to error. Make config envvar accept both `vidcapture` and `videocapture` (case insensitive) and fix `get_backends` returning `VidCapture` when it should be `VideoCapture`, and some more code cleanups around that. Also errors with a more helpful message if a particular backend is unavailable

### CODE REVIEW

**Comment by Starbuck5** [comment]
Thanks Ankith, but yeesh, this diff is something else. This changes 110 lines of a <200 line file.

I don't want to expand environment variables for vidcapture, because that thing is 100% deprecated. In fact, in my last camera PR, illume told me I could remove it, but I kept it, in a role reversal for the ages. Actually, vidcapture could use the deprecation warning. The biggest problem for me is that I've **never** been able to get it working, even though there are third party python 3 wheels (vidcapture is py2) that supposedly work.

**Comment by Starbuck5** [comment]
Ok, I've read through everything, it looks good.

I'm curious why you refactored all of my individual setup functions into one, though.

**Comment by ankith26** [comment]
> I'm curious why you refactored all of my individual setup functions into one, though.

Those functions had some unnecessary repetition, and this one function conditional approach seems more elegant to me than having functions kept in a dict and each of those functions having repetition of  `global` statement and `_camera` import 😄 

> Maybe colorspace should have a different error message if it isn't able to be imported. Which it could be, if pygame was built without the _camera module. I don't think it's built on Windows 7.

A `NotImplementedError` is the most pythonic error to be using in a case like this, it was made for a case like this.

> I don't want to expand environment variables for vidcapture, because that thing is 100% deprecated. In fact, in my last camera PR, illume told me I could remove it, but I kept it, in a role reversal for the ages.

Ah, ok. Should I revert my change then? Well it does not matter either way IG, this vidcapture changes should remain undocumented anyways

Another thing I wanted to ask, does windows `_camera` work on win8? Because the code does `int(platform.win32_ver()[0]) > 8` check, should it be `>=`?

**Comment by Starbuck5** [comment]
It should work on Windows 8, since every function I used has been around since then, but I’m still not sure. It claims to support it in the docs, so we should try to use it in the code. If it doesn’t work, somebody will open an issue. (Or nobody will care, since not many people use pygame.camera and not many people use Windows 8)

**Comment by ankith26** [comment]
Alrighty :)
I also updated the `colorspace` function to emit better errors when not available, although yeah, it should not get triggered in most cases

File: src_py/camera.py [hunk 1]
```diff
@@ -1,82 +1,132 @@
...
+    if not _is_init:
+        raise error("pygame.camera is not initialized")
 
-    list_cameras = _camera_opencv.list_cameras
-    Camera = _camera_opencv.Camera
-    if _camera:
-        colorspace = _camera.colorspace
+    # camera was init, and yet functions are not monkey patched. This should
+    # not happen
+    raise NotImplementedError()
```
Comment (Starbuck5) [inline src_py/camera.py:56]: is this supposed to be here?

File: src_py/camera.py [hunk 2]
```diff
@@ -1,82 +1,132 @@
...
 
+_is_init = False
 
-def _setup_opencv_mac():
-    global list_cameras, Camera, colorspace
 
-    from pygame import _camera_opencv
+class AbstractCamera(ABC):
+    @abstractmethod
+    def __init__(self, device=0, size=(320, 200), mode="RGB"):
```
Comment (Starbuck5) [inline src_py/camera.py:14]: default size is (640, 480)

File: src_py/camera.py (hunk 1 above)
Comment (ankith26) [reply src_py/camera.py:56]: Yeah, just for extra safety, IG. I can't think of a case where this codepath will actually be executed

File: src_py/camera.py (hunk 2 above)
Comment (ankith26) [reply src_py/camera.py:14]: ooh, nice catch!

**Review by Starbuck5 — APPROVED** [review]
Thanks, Ankith!

This looks good to me. You could fix that tiny remaining thing (the default size is wrong on the class), and the it'll be good.

**Review by illume — APPROVED** [review]
👍 🎉 thanks

**Pull request merged**